.github/workflows/unit-tests.yml

```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - main
+      - feature/dep_cache
```

Comment:
Don't forget to remove this once you're done.

Reply:
> Really cool! Can you add the same to the `branch-docs`, `dev-docs`, and `branch-tests` jobs?
Now that I think about adding the same step, these jobs will use the same cache any way, and if a cache doesn't exist, they will each create their own environments and overwrite each other's.

So, how would you feel about simplifying these jobs, by moving the environment creation to a separate job, on which `unit-tests`, `branch-docs`, `dev-docs`, and `branch-tests`? This should be possible with GitHub actions ([blog post](https://github.blog/2021-11-18-7-advanced-workflow-automation-features-with-github-actions/#2-build-breakpoints-in-a-workflow-with-dependencies-or-create-dependencies-between-workflows)).

This could make the jobs easier to maintain (only one place to update) and downstream jobs very fast (although the end-to-end duration would remain the same).

In fact, we could extend this even to the river build step, so only one job would call `pip install -e ".[dev]"`.

Reply:
I agree that GitHub Actions is the way to go, I'm glad you mention it :)